run/shas_test.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+#
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import mock
+import os
+import shas
+import unittest
+
+from datetime import date
+
+here = os.path.dirname(__file__)
+wptd_dir = os.path.join(here, '../')
+
+
+class TestSHAFinder(unittest.TestCase):
+
+    def test_nov_21st(self):
+        logger = mock.Mock(logging.Logger)
+
+        sha_finder = shas.SHAFinder(logger, date(2017, 11, 21))
+
+        self.assertEqual('46060eb2c33de6101bc6930bf5e34f794aa9f996',
```

Comment:
Trying to find another commit to test with, I think I'm wrong about `--date-order`. Removing it doesn't make any difference in all commits since 2017-01-01, but adding `--topo-order` on the other hand does.

Just one test without the edge case is fine, but I guess if you wanted to find such a case in history, the easiest way would be to loop over dates here until you find one where adding `--topo-order` gives a different result, and picking which is the best behavior.